### ISSUE TITLE
Is there a way to Automatically Remove Featured Artists from Song Title

### ISSUE THREAD

**Issue by thyWyrm** [yt-dlp/yt-dlp#9010]
### DO NOT REMOVE OR SKIP THE ISSUE TEMPLATE

- [X] I understand that I will be **blocked** if I *intentionally* remove or skip any mandatory\* field

### Checklist

- [X] I'm asking a question and **not** reporting a bug or requesting a feature
- [X] I've looked through the [README](https://github.com/yt-dlp/yt-dlp#readme)
- [X] I've verified that I have **updated yt-dlp to nightly or master** ([update instructions](https://github.com/yt-dlp/yt-dlp#update-channels))
- [X] I've searched [known issues](https://github.com/yt-dlp/yt-dlp/issues/3766) and the [bugtracker](https://github.com/yt-dlp/yt-dlp/issues?q=) for similar questions **including closed ones**. DO NOT post duplicates
- [X] I've read the [guidelines for opening an issue](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#opening-an-issue)

### Please make sure the question is worded well enough to be understood

Hi all,

I'm currently using yt-dlp to download albums from youtube, as it automatically fills in all the relevant metadata fields. The only problem is that when there's a featured artist in the title, it keeps the featured name in the title.

Example: Plastic Beach Album by Gorillaz
Current Title: Orchestral Intro (feat. Sinfonia ViVA)
What I'd Like to get: Orchestral Intro

1) Would there be a way to automatically remove " (feat. artist)" from the Title of a song? I was thinking I could remove all the text to the right starting at " (feat.", but I'm not sure how to do that. Is there a way to automate such a process?

Here's the list of commands I'm currently using:

`yt-dlp --extract-audio --audio-format mp3 --embed-metadata --parse-metadata "playlist_index:%(track_number)s" --add-metadata -o "../Music/<FOLDER NAME>/%(title)s.%(ext)s" --parse-metadata ":(?P<meta_date>)" <SONG URL>`

2) While we are at it, would there be a way for me to indicate which release year I want to use in my metadata? Currently, I'm using ` --parse-metadata ":(?P<meta_date>)"` to remove the included youtube video upload date. Would there be a way to manually include the date in my command?

Thanks in advance for the help.

### Provide verbose output that clearly demonstrates the problem

- [X] Run **your** yt-dlp command with **-vU** flag added (`yt-dlp -vU <your command line>`)
- [X] If using API, add `'verbose': True` to `YoutubeDL` params instead
- [X] Copy the WHOLE output (starting with `[debug] Command-line config`) and insert it below

### Complete Verbose Output

```shell
yt-dlp --extract-audio --audio-format mp3 --embed-metadata --parse-metadata "playlist_index:%(track_number)s" --add-metadata -o "../Music/Plastic Beach/%(title)s.%(ext)s" --parse-metadata ":(?P<meta_date>)" https://music.youtube.com/playlist?list=OLAK5uy_lrJakDq8vTWmFU327tp_oI1zgSishISOY
[youtube:tab] Extracting URL: https://music.youtube.com/playlist?list=OLAK5uy_lrJakDq8vTWmFU327tp_oI1zgSishISOY
WARNING: [youtube:tab] YouTube Music is not directly supported. Redirecting to https://www.youtube.com/playlist?list=OLAK5uy_lrJakDq8vTWmFU327tp_oI1zgSishISOY
[youtube:tab] OLAK5uy_lrJakDq8vTWmFU327tp_oI1zgSishISOY: Downloading webpage
[youtube:tab] OLAK5uy_lrJakDq8vTWmFU327tp_oI1zgSishISOY: Redownloading playlist API JSON with unavailable videos
[download] Downloading playlist: Album - Plastic Beach
[youtube:tab] Playlist Album - Plastic Beach: Downloading 16 items of 16
[download] Downloading item 1 of 16
[youtube] Extracting URL: https://music.youtube.com/watch?v=PSffezqR_XA
[youtube] PSffezqR_XA: Downloading webpage
[youtube] PSffezqR_XA: Downloading ios player API JSON
[youtube] PSffezqR_XA: Downloading android player API JSON
[youtube] PSffezqR_XA: Downloading ios music player API JSON
[youtube] PSffezqR_XA: Downloading android music player API JSON
[youtube] PSffezqR_XA: Downloading web music client config
[youtube] PSffezqR_XA: Downloading player 80b90bfd
[youtube] PSffezqR_XA: Downloading web music player API JSON
[youtube] PSffezqR_XA: Downloading m3u8 information
[youtube] PSffezqR_XA: Downloading m3u8 information
[MetadataParser] Parsed track_number from '%(playlist_index)s': '01'
[MetadataParser] Parsed meta_date from '': ''
[info] PSffezqR_XA: Downloading 1 format(s): 251
[download] Destination: ..\Music\Plastic Beach\Orchestral Intro (feat. Sinfonia ViVA).webm
[download] 100% of    1.18MiB in 00:00:00 at 8.54MiB/s
[ExtractAudio] Destination: ..\Music\Plastic Beach\Orchestral Intro (feat. Sinfonia ViVA).mp3
Deleting original file ..\Music\Plastic Beach\Orchestral Intro (feat. Sinfonia ViVA).webm (pass -k to keep)
[Metadata] Adding metadata to "..\Music\Plastic Beach\Orchestral Intro (feat. Sinfonia ViVA).mp3"

THEN IT MOVES ON TO THE NEXT SONG AND REPEATS
```

### UPDATED VERBOSE OUTPUT

```
yt-dlp --extract-audio --audio-format mp3 --embed-metadata --parse-metadata "playlist_index:%(track_number)s" --add-metadata -o "../Music/AAA/%(title)s.%(ext)s" --parse-metadata "title:(?P<title>.+)\s+\(f(?:ea)?t(?:uring|\.)?" --parse-metadata " 2010: %(meta_date)s" --add-metadata https://music.youtube.com/playlist?list=OLAK5uy_lrJakDq8vTWmFU327tp_oI1zgSishISOY
[youtube:tab] Extracting URL: https://music.youtube.com/playlist?list=OLAK5uy_lrJakDq8vTWmFU327tp_oI1zgSishISOY
WARNING: [youtube:tab] YouTube Music is not directly supported. Redirecting to https://www.youtube.com/playlist?list=OLAK5uy_lrJakDq8vTWmFU327tp_oI1zgSishISOY
[youtube:tab] OLAK5uy_lrJakDq8vTWmFU327tp_oI1zgSishISOY: Downloading webpage
[youtube:tab] OLAK5uy_lrJakDq8vTWmFU327tp_oI1zgSishISOY: Redownloading playlist API JSON with unavailable videos
[download] Downloading playlist: Album - Plastic Beach
[youtube:tab] Playlist Album - Plastic Beach: Downloading 16 items of 16
[download] Downloading item 1 of 16
[youtube] Extracting URL: https://music.youtube.com/watch?v=PSffezqR_XA
[youtube] PSffezqR_XA: Downloading webpage
[youtube] PSffezqR_XA: Downloading ios player API JSON
[youtube] PSffezqR_XA: Downloading android player API JSON
[youtube] PSffezqR_XA: Downloading ios music player API JSON
[youtube] PSffezqR_XA: Downloading android music player API JSON
[youtube] PSffezqR_XA: Downloading web music client config
[youtube] PSffezqR_XA: Downloading player 80b90bfd
[youtube] PSffezqR_XA: Downloading web music player API JSON
[youtube] PSffezqR_XA: Downloading m3u8 information
[youtube] PSffezqR_XA: Downloading m3u8 information
[MetadataParser] Parsed track_number from '%(playlist_index)s': '01'
[MetadataParser] Parsed title from '%(title)s': 'Orchestral Intro'
[MetadataParser] Parsed meta_date from ' 2010': '2010'
[info] PSffezqR_XA: Downloading 1 format(s): 251
[download] Destination: ..\Music\AAA\Orchestral Intro.webm
[download] 100% of    1.18MiB in 00:00:00 at 3.31MiB/s
[ExtractAudio] Destination: ..\Music\AAA\Orchestral Intro.mp3
Deleting original file ..\Music\AAA\Orchestral Intro.webm (pass -k to keep)
[Metadata] Adding metadata to "..\Music\AAA\Orchestral Intro.mp3"
```

**Comment by bashonly** [comment]
could try something like this
`--parse-metadata "title:(?P<title>.+)\s+\(f(?:ea)?t(?:uring|\.)?"`

**Comment by thyWyrm** [comment]
Thanks for the help, I really appreciate it!

I've tried using `--parse-metadata "title:(?P<title>.+)\s+\(f(?:ea)?t(?:uring|\.)?"` in my command, and there seems to be a spot in the verbose output (I added the updated version above) where it correctly isolates the title, but for some reason it doesn't give that value to the mp3 once the song is downloaded. The song title still has the (feat. Sinfonia ViVA). Not sure what's happening there. I've tried calling the command at different places but it doesn't seem to make a difference.

Updated command:
`yt-dlp-master>yt-dlp --extract-audio --audio-format mp3 --embed-metadata --parse-metadata "playlist_index:%(track_number)s" --add-metadata -o "../Music/AAA/%(title)s.%(ext)s" --parse-metadata "title:(?P<title>.+)\s+\(f(?:ea)?t(?:uring|\.)?" --parse-metadata " 2010: %(meta_date)s" https://music.youtube.com/playlist?list=OLAK5uy_lrJakDq8vTWmFU327tp_oI1zgSishISOY`

The one for the date works like a charm thanks :) !!
